File: modules/simpleServer/src/main/scala/org/validoc/simpleServer/SimpleHttpServer.scala
package org.validoc.simpleServer

import java.net.InetSocketAddress
import java.util.concurrent.Executor

import com.sun.net.httpserver.HttpServer

class SimpleHttpServer (port: Int, executor: Executor, handlers: PathAndHandler*){
  private val server = HttpServer.create(new InetSocketAddress(port), 0)

  handlers.foreach(server.createContext())
  def this {
    this()
    this.server =
    for (pathAndHandler <- handlers) {
      server.createContext(pathAndHandler.path, pathAndHandler)
    }
    server.setExecutor(executor)
  }

  def start(): Unit = {
    server.start()
  }

  def stop(): Unit = {
    server.stop(0)
  }

}

File: modules/cddexamples/src/main/scala/core/SimpleHttpServer.java
package core;


import com.sun.net.httpserver.HttpServer;

import java.io.IOException;
import java.net.InetSocketAddress;
import java.util.concurrent.Executor;

public class SimpleHttpServer {

    private final HttpServer server;

    public SimpleHttpServer(int port, Executor executor, PathAndHandler... handlers) throws IOException {
        this.server = HttpServer.create(new InetSocketAddress(port), 0);
        for (PathAndHandler pathAndHandler : handlers) {
            server.createContext(pathAndHandler.path(), pathAndHandler);
        }
        server.setExecutor(executor);
    }

    public void start() {
        server.start();
    }

    public void stop() {
        server.stop(0);
    }


}
